packages/mantine/src/components/header/Header.tsx

```diff
@@ -92,7 +90,7 @@ export interface HeaderDocAnchorProps {
 
 const HeaderDocAnchor: FunctionComponent<HeaderDocAnchorProps> = ({href: docLink, label: docLinkTooltipLabel}) => (
     <Tooltip label={docLinkTooltipLabel} disabled={!docLinkTooltipLabel} position="right">
-        <Anchor inline href={docLink} target="_blank" ml="xs">
+        <Anchor inline href={docLink} target="_blank" ml="xs" style={{verticalAlign: 'middle'}}>
```

Comment:
This aligns the top of the icon to the middle of the text

<img width="211" alt="image" src="https://user-images.githubusercontent.com/35579930/232787337-fbe4ed64-df12-49ba-b057-f44a9fa130a9.png">

What we want is align the middle of the icon with the middle of the text. To do that here I used a flex aligned on the center on the h1 element

<img width="165" alt="image" src="https://user-images.githubusercontent.com/35579930/232787767-b3c40279-03e9-4bbc-b383-d9d7b479e8cf.png">

Reply:
I validated the new alignment with our UX

The `Title` example is flawed because there's no descender letter, but it is centered

<img width="199" alt="image" src="https://user-images.githubusercontent.com/260007/232792568-9bf8ea57-7e85-49ae-8590-8b4ba779d248.png">